app/views/AnalyticalFramework/PrimaryTagging/index.tsx

```diff
@@ -127,8 +128,11 @@ function PrimaryTaggingInput<K extends string>(props: PrimaryTaggingInput<K>) {
         (value: Section[]) => {
             setTempSections(undefined);
             setSections(value, name);
+            if (sectionToEdit) {
+                setSelectedSection(sectionToEdit);
+            }
```

Comment:
We should not use sectionToEdit to select selectionSection.

On each section change, we should see if selectedSection is removed.
If it's removed, we should set other section as selected (can be the first or the nearest element)

We should also do this check on section cancel, which should restore the previous section if currently selected section is no longer there.